main.go

```diff
@@ -56,42 +62,27 @@ func main() {
 	flags := parseFlags()
 
 	// Prepare output directory if necessary
-	if flags.OutputDir != flags.InputDir {
-		if err := clearAndCreateDir(flags.OutputDir); err != nil {
-			log.Fatal("Error preparing output directory:", err)
-		}
+	if flags.Mode == modeBundle && flags.OutputDir != flags.InputDir {
+		prepareOutputDirectory(flags.OutputDir)
 	}
 
 	// Prepare directories and find files
 	modulePath, groupedFiles := prepareDirectories(flags.InputDir)
 
 	// Run templ fmt if specified
-	if flags.RunFormat {
-		runTemplFmt(groupedFiles)
-	}
+	runTemplFmtIfNeeded(flags.RunFormat, groupedFiles)
 
 	// Run templ generate if specified
-	if flags.RunGenerate {
-		groupedFiles = runTemplGenerate(flags.InputDir)
-	}
+	groupedFiles = runTemplGenerateIfNeeded(flags.RunGenerate, flags.InputDir)
```

Comment:
this helper function results in a nil pointer dereference if no `-g` flag is provided, since it return nil if the flag is false. Replace these helper fucntions with simple if statements here, instead of run...ifNeeded

I'm assuming you are trying to keep the cyclomatic complexity low, but I'd argue its better to keep the logic simple and "inline", rather than moving it to many helpers and potentially introduce bugs in the process.

Also, I believe the call to `runTemplGenerate` could be done before `finder.FindFilesInDir` (which returns the `groupedFiles`), this way you don't end up calling `finder.FindFilesInDir` twice.

Reply:
My goal is not keep the cyclomatic complexity low forcing design :) will do some rework